templating/templating.py

```
@@ -8,17 +8,20 @@
 import requests
 
 from flask import Flask, request
-from jinja2 import Environment, FileSystemLoader, TemplateNotFound
+from jinja2 import Environment, FileSystemLoader, TemplateNotFound, TemplateError
 
 endpoints = "read/networks read/oplog read/snmp read/switches-management public/distro-tree public/config public/dhcp public/dhcp-summary public/ping public/switches public/switch-state".split()
 
 objects = {}
 
 
 def getEndpoint(endpoint):
-    r = requests.get("http://localhost:80/api/{}".format(endpoint))
-    if r.status_code != 200:
-        raise Exception("Bad status code for endpoint {}: {}".format(endpoint, r.status_code))
+    uri = "{}/api/{}".format(args.server, endpoint)
+    try:
+        r = requests.get(uri, timeout=args.timeout)
```

Common to do the `r.raise_for_status` in the try-except, but I think the exception catching is a bit redundant, since you don't do anything but re-raising. If there is an exception, let it just raise up to the function that calls it, unless you're actually actively doing something with that exception.  

I'd probably rewrite this entire function to look something like this: 
```py
def getEndpoint(endpoint: str) -> dict:
    """
    Fetches an endpoint and returns the data as a dict.
    """
    uri = f"{args.server}/api/{endpoint}"
    r = requests.get(uri, timeout=args.timeout)
    r.raise_for_status()
    return r.json()
```

and in the function that calls `getEndpoint` (e.g. `root_get())` you catch it like you do here:
```py
    except (requests.exceptions.ConnectTimeout, requests.exceptions.ConnectionError) as error:
        return f'Timeout or connection error from gondul: {err}', 500
```